### PR TITLE
COMMERCE-11790

### PR DESCRIPTION
**Pull request by brianikim**
@andrea-ale-sbarra ,

https://liferay.atlassian.net/browse/COMMERCE-11790

### CODE REVIEW

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/search/CPDefinitionSearcher.java [hunk 1]
```diff
@@ -51,14 +51,23 @@ public CPDefinitionSearcher(CPQuery cpQuery) {
...
+
+	@Override
+	public String getClassName() {
+		return _CLASS_NAME;
 	}
 
 	@Override
 	public String[] getSearchClassNames() {
-		return new String[] {CPDefinition.class.getName()};
+		return new String[] {_CLASS_NAME};
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/search/CPDefinitionSearcher.java:70]: https://github.com/liferay-commerce/liferay-portal/pull/4003#discussion_r1261118654

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/search/spi/model/query/contributor/CPDefinitionModelPreFilterContributor.java [hunk 2]
```diff
@@ -48,26 +57,236 @@ public void contribute(
...
 		SearchContext searchContext) {
 
+		booleanFilter.addRequiredTerm(Field.HIDDEN, false);
+
+		_filterByBasePrice(booleanFilter, searchContext);
+		_filterByDefinitionLinks(booleanFilter, searchContext);
+		_filterByExcludedCPDefinitionId(booleanFilter, searchContext);
+		_filterByPublished(booleanFilter, searchContext);
+		_filterByStatuses(booleanFilter, searchContext);
+		_filterBySubscriptionEnabled(booleanFilter, searchContext);
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/search/spi/model/query/contributor/CPDefinitionModelPreFilterContributor.java:67]: https://github.com/liferay-commerce/liferay-portal/pull/4003#discussion_r1261142990

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/search/spi/model/result/contributor/CPDefinitionModelSummaryContributor.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+	implements ModelSummaryContributor {
+
+	@Override
+	public Summary getSummary(
+		Document document, Locale locale, String snippet) {
+
+		String prefix = Field.SNIPPET + StringPool.UNDERLINE;
+
+		String title = document.get(prefix + Field.NAME, Field.NAME);
+		String content = document.get(
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/search/spi/model/result/contributor/CPDefinitionModelSummaryContributor.java:44]: please make sure this is also fixing this https://liferay.atlassian.net/browse/COMMERCE-11771

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/search/spi/model/query/contributor/CPInstanceModelPreFilterContributor.java [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+		BooleanFilter booleanFilter, ModelSearchSettings modelSearchSettings,
+		SearchContext searchContext) {
+
+		_filterByCPDefinitionId(booleanFilter, searchContext);
+		_filterByCPDefinitionStatus(booleanFilter, searchContext);
+		_filterByHasChildDefinitions(booleanFilter, searchContext);
+		_filterByOptions(booleanFilter, searchContext);
+		_filterByPublished(booleanFilter, searchContext);
+		_filterByPurchasable(booleanFilter, searchContext);
+		_filterByStatus(booleanFilter, searchContext);
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/search/spi/model/query/contributor/CPInstanceModelPreFilterContributor.java:50]: same question here

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/search/spi/model/index/contributor/CPInstanceModelDocumentContributor.java [hunk 5]
```diff
@@ -105,6 +107,20 @@ public void contribute(Document document, CPInstance cpInstance) {
...
 
+			List<CPInstanceUnitOfMeasure> cpInstanceUnitOfMeasures =
+				cpInstance.getCPInstanceUnitOfMeasures(
+					QueryUtil.ALL_POS, QueryUtil.ALL_POS, null);
+
+			if (!cpInstanceUnitOfMeasures.isEmpty()) {
+				for (CPInstanceUnitOfMeasure cpInstanceUnitOfMeasure :
+						cpInstanceUnitOfMeasures) {
+
+					document.addKeyword(
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/search/spi/model/index/contributor/CPInstanceModelDocumentContributor.java:118]: I would add the unitOfMeasure key too.

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/search/spi/model/query/contributor/CPDefinitionModelPreFilterContributor.java (hunk 2 above)
Comment (brianikim) [reply modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/search/spi/model/query/contributor/CPDefinitionModelPreFilterContributor.java:67]: Direct port of `postProcessContextBooleanFilter`: https://github.com/liferay-commerce/liferay-portal/blob/master/modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/search/CPDefinitionIndexer.java#L124

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/search/spi/model/result/contributor/CPDefinitionModelSummaryContributor.java (hunk 3 above)
Comment (brianikim) [reply modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/search/spi/model/result/contributor/CPDefinitionModelSummaryContributor.java:44]: I will fix as part of this ticket, but the solution for that ticket is not tied to search @andrea-ale-sbarra .

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
ci:test:commerce-functional

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
ci:test:commerce-functional

**Review by luca-pellizzon — COMMENTED** [review]
LGTM, but let's see the results of the tests

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Comment by luca-pellizzon** [comment]
ci:test:commerce-functional

**Comment by brianikim** [comment]
ci:test:commerce-functional

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
ci:test:commerce-functional

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
ci:test:commerce-functional

**Comment by brianikim** [comment]
@luca-pellizzon , failures are legitimate. Investigating. See COMMERCE ticket for more details.

**Pull request closed without merging**